khelo/lib/ui/flow/tournament/detail/members/tournament_detail_members_screen.dart

```diff
@@ -169,106 +172,88 @@ class _TournamentDetailMembersScreenState
       return;
     }
 
-    List<BottomSheetAction> actions = [];
     bool isOwner = currentUserId == tournament.created_by;
+    bool isSelf = currentUserId == member.id;
 
-    if (currentUserId == member.id && !isOwner) {
-      actions.add(
-        BottomSheetAction(
-          title: context.l10n.tournament_members_remove_self,
-          onTap: () async {
-            context.pop();
-            notifier.removeTournamentMember(tournament.id, member);
-          },
-        ),
-      );
+    List<TournamentMemberActionType> availableActions = [];
+
+    if (isSelf && !isOwner) {
+      availableActions.add(TournamentMemberActionType.removeSelf);
     }
 
     if (isOwner) {
-      if (currentUserId == member.id) {
-        actions.add(
-          BottomSheetAction(
+      if (isSelf) {
+        availableActions.add(TournamentMemberActionType.transferOwnership);
+      } else if (member.role == TournamentMemberRole.admin) {
+        availableActions.add(TournamentMemberActionType.makeOrganizer);
+        availableActions.add(TournamentMemberActionType.removeMember);
+      } else {
+        availableActions.add(TournamentMemberActionType.makeAdmin);
+        availableActions.add(TournamentMemberActionType.removeMember);
+      }
+    } else if (currentUserRole == TournamentMemberRole.organizer) {
+      if (member.role != TournamentMemberRole.admin) {
+        availableActions.add(TournamentMemberActionType.makeAdmin);
+        availableActions.add(TournamentMemberActionType.removeMember);
+      }
+    }
+
+    List<BottomSheetAction> actions = availableActions.map((actionType) {
+      switch (actionType) {
```

Comment:
no need for switch and separate returning of a BottomSheetAction, just take string from **enum.getString(context)** and call **notifier.function(TournamentMemberActionType)** which will handle action based on type.